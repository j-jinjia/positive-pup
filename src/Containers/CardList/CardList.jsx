import ServicesCard from "../../components/ServicesCard/ServicesCard";
import "./CardList.scss";

const servicesCardList = ({ cardData }) => {
  const cardsListJSX = cardData.map((card, index) => {
    return (
      <ServicesCard
        key={index}
        title={card.title}
        description={card.description}
        image={card.image}
      />
    );
  });
  return <div className="services-card-list">{cardsListJSX}</div>;
};

export default servicesCardList;
