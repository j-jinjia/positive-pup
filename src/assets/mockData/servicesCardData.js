import image0 from "../mockData/images/image.png";
import image1 from "../mockData/images/image1.png";
import image2 from "../mockData/images/image2.png";

const courses = [
  {
    id: 0,
    image: image0,
    title: "Online Courses",
    description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit.",
  },
  {
    id: 1,
    image: image1,
    title: "Group Courses",
    description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit.",
  },
  {
    id: 2,
    image: image2,
    title: "Bespoke Packages",
    description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit.",
  },
];

export default courses;
