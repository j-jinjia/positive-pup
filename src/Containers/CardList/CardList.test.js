import { customRender } from "../../utils/testUtils";
import CardList from "./CardList";
import servicesCardData from "../../assets/mockData/courseData";

it("should match CardList component snapshot", () => {
  const { container } = customRender(<CardList cardData={servicesCardData} />);

  expect(container).toMatchSnapshot();
});
