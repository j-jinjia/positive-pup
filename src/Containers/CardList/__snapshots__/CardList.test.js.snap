// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match CardList component snapshot 1`] = `
<div>
  <div
    class="card-list"
  >
    <div
      class="services-card"
    >
      <h3
        class="services-card__title"
      >
        Online Courses
      </h3>
      <p
        class="services-card__text"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
      <a
        class="services-card__button"
        href="/services/Online Courses"
      >
        <img
          alt=""
          class="services-card__vector services-card__vector--black"
          src="Vector-black.svg"
        />
      </a>
      <img
        alt="dog"
        class="services-card__img"
        src="image.png"
      />
    </div>
    <div
      class="services-card"
    >
      <h3
        class="services-card__title"
      >
        Group Classes
      </h3>
      <p
        class="services-card__text"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
      <a
        class="services-card__button"
        href="/services/Group Classes"
      >
        <img
          alt=""
          class="services-card__vector services-card__vector--black"
          src="Vector-black.svg"
        />
      </a>
      <img
        alt="dog"
        class="services-card__img"
        src="image1.png"
      />
    </div>
    <div
      class="services-card"
    >
      <h3
        class="services-card__title"
      >
        Bespoke Packages
      </h3>
      <p
        class="services-card__text"
      >
        Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      </p>
      <a
        class="services-card__button"
        href="/services/Bespoke Packages"
      >
        <img
          alt=""
          class="services-card__vector services-card__vector--black"
          src="Vector-black.svg"
        />
      </a>
      <img
        alt="dog"
        class="services-card__img"
        src="image4.png"
      />
    </div>
  </div>
</div>
`;
